core/src/main/java/org/geysermc/geyser/inventory/GeyserItemStack.java

```diff
@@ -122,7 +126,7 @@ public int getAmount() {
      * @return the item's base data components and the "additional" ones that may exist.
      */
     public @Nullable DataComponents getAllComponents() {
```

Comment:
How practical/reasonable would it be to remove the `componentCache` field and replace this with a `getAllComponents(GeyserSession)` call? Having all items defined with a session parameter still rubs me the wrong way. Defining `EMPTY` above with a null session solidifies that.